pkg/config/datafileprojectconfig/mappers/condition_trees.go

```diff
@@ -93,22 +81,64 @@ func buildConditionTree(conditions interface{}) (conditionTree *entities.TreeNod
 		}
 	}
 
-	populateConditions(value, conditionTree)
+	// Check for leaf conditions
+	if value.Kind() == reflect.Map {
+		typedV := value.Interface()
+		if v, ok := typedV.(map[string]interface{}); ok {
+			n := &entities.TreeNode{}
+			if err := createLeafCondition(v, n); err != nil {
+				retErr = err
+				return
+			}
+			conditionTree.Operator = "or"
+			conditionTree.Nodes = append(conditionTree.Nodes, n)
+		}
+	} else {
+		populateConditions(value, conditionTree)
+	}
+
 	if conditionTree.Nodes == nil && conditionTree.Operator == "" {
 		retErr = errEmptyTree
 		conditionTree = nil
 	}
 	return conditionTree, retErr
 }
 
+// Validates conditions for audience in the datafile
+func validateConditions(conditions interface{}) (parsedConditions interface{}, retErr error) {
+	switch v := conditions.(type) {
```

Comment:
This doesn’t really validate conditions. It’s just trying to parse it into an interface

Reply:
yeah, maybe the name is not meaningful, but it just parses this into interface

Reply:
changed method name to `parseConditions`.